### ISSUE TITLE
`reportDeprecated` broken in 1.1.341

### ISSUE THREAD

**Issue by ottumm** [microsoft/pyright#6754]
Starting in 1.1.341, usage of deprecated functions isn't caught by Pyright (CLI). The code sample below shows this, and switching to 1.1.340 reports an error as expected.

Code sample in [pyright playground](https://pyright-play.net/?strict=true&code=GYJw9gtgBALgngBwJYDsDmB9ApgDxllAZyTCKiQgTBBigBMsEQsBjAQ3zoCgeABBpqw5Y6ACgBEAEVIByWgFdCWWAAskhcQBoo7fGmpwAvADlSWAJRcGwKMDBhR5gFxcobqADovPLnYeWgA)

```python
from typing_extensions import deprecated


@deprecated("Don't use this", category=None)
def foo():
    ...


foo()

```

**Comment by erictraut** [comment]
Yes, I already noticed and fixed the problem. It was caused by an unexpected change that was recently made in the typeshed stubs. The `deprecated` symbol was previously defined as a function, but it was changed to a class. 

This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.342, which I just published. It will also be included in a future release of pylance.
